e2e-webdriver/src/page-objects/app.po.ts

```diff
@@ -0,0 +1,18 @@
+export class AppPage {
+
+  async navigateTo(): Promise<string> {
+    return browser.url('');
+  }
+
+  async getTitle(): Promise<string> {
+    const $h1: WebdriverIO.Element = await $('app-root h1');
+    return $h1.getText();
+  }
+
+  async getProductLinkByName(name: string): Promise<WebdriverIO.Element> {
+    const $productMaster = await $('app-root app-product-master');
+    const $name = await $productMaster.$(`.product-title=${name}`);
```

Comment:
Die zwei Selektoren kannst du zusammenziehen. Das spart nochmal einen Request.
```
const productName = await $(`app-root app-product-master .product-title=${name}`);

```

Reply:
Das funktioniert leider nicht so. Steht auch in der Doku:

> Note: You can't mix multiple selector strategies in one selector. Use multiple chained element queries to reach the same goal, e.g.:

```
const elem = $('header h1*=Welcome') // doesn't work!!!
// use instead
const elem = $('header').$('*=driver')
```

Reply:
Da hast du Recht.